### ISSUE TITLE
How to get to generated input element

### ISSUE THREAD

**Issue by redhead** [danialfarid/ng-file-upload#1140]
We are using custom div element as an `ngf-select` as well as a `data-ngf-drop`. How can I get to the `input[type=file]` element generated by the directive? We need to add an ID to it so we can upload files in automated tests.

Could it be possible to add a reference to the generated element to the ngf-select element (in a property) or better, make a directive controller that would expose the element as part of it's API? This way I can get to the element in a wrapping directive, so I can add the ID.

**Comment by danialfarid** [comment]
You can search for input type file with ngf-select appended at the end of the body or if you have multiple inputs you can set an arbitrary attribute like data-ngf-id="myId" for the div and that will be copied over to the generated input type file.

**Comment by redhead** [comment]
Yes I went with the ID data tribute the first time. The thing is that the ID is generated, and when I use `{{id}}` then it gets copied with the braces and not with the actual value of `id` property. I had to create a directive that adds the attribute dynamically in pre-link function based on a scope property.

I think, it would be better to actually expose some API through a controller (that's its purpose).
